tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,321 @@
+  afterAll(async () => {
+    await pkAgentClose();
+  });
+  beforeEach(async () => {
     dataDir = await fs.promises.mkdtemp(
       path.join(os.tmpdir(), 'polykey-test-'),
     );
-    nodePath = path.join(dataDir, 'keynode');
-    command = ['vaults', 'list', '-np', nodePath];
-    passwordFile = path.join(dataDir, 'passwordFile');
-    sessionFile = path.join(nodePath, 'token');
-    await fs.promises.writeFile(passwordFile, 'password');
-    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
-      password: 'password',
-      nodePath: nodePath,
-      logger: logger,
+    // Invalidate all active sessions
+    await testBinUtils.pkStdio(
+      ['agent', 'lock'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    );
+  });
+  afterEach(async () => {
+    await fs.promises.rm(dataDir, {
+      force: true,
+      recursive: true,
     });
-    await polykeyAgent.vaultManager.createVault(vaultName);
   });
-
-  afterAll(async () => {
-    await polykeyAgent.stop();
-    await polykeyAgent.destroy();
-    await fs.promises.rmdir(dataDir, { recursive: true });
+  test('processes should store session token in session file', async () => {
+    // Run command to set token
+    let exitCode, stdout;
+    ({ exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--verbose'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    ));
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain('LIVE');
+    await sleep(1100);
+    // Try again without password
+    ({ exitCode, stdout } = await testBinUtils.pkStdio(
+      ['agent', 'status', '--verbose'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+      },
+      global.binAgentDir,
+    ));
+    expect(exitCode).toBe(0);
+    expect(stdout).toContain('LIVE');
   });
-
-  test('Process should store session token in session file', async () => {
-    // Agent has not been unlocked yet
-    const result = await testUtils.pkStdio(
-      command,
-      { PK_PASSWORD: 'password' },
-      dataDir,
-    );
-    expect(result.exitCode).toBe(0);
-    expect(result.stdout).toContain(vaultName);
-
-    const buff = await fs.promises.readFile(sessionFile);
-    const newToken = buff.toString() as SessionToken;
-    expect(
-      async () => await polykeyAgent.sessionManager.verifyToken(newToken),
-    ).not.toThrow();
+  test('processes should refresh the session token', async () => {
+    const session = await Session.createSession({
+      sessionTokenPath,
+      fs,
+      logger,
+    });
+    // Generate new token
+    await testBinUtils.pkStdio(
+      ['agent', 'unlock'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    );
+    await sleep(1100);
+    const token1 = await session.readToken();
```

Comment:
Perhaps instead, for this instance, we can `poll` on `readToken`?

Reply:
Unless `readToken` is expected to always be defined. My assumption here was that the sleep was here such that we give it time for the token to be created (i.e. `readToken` transitions from returning `undefined` to some returned token after `pk agent unlock`).

Reply:
If we expect a change in the token, you could also just store the initial token, and `poll` until the token is different.

Reply:
Gonna play with this.

Reply:
No need to poll. Just use `status.waitFor`. See how in other tests we do this.

Reply:
Oh sorry this is token...

Reply:
Resolved by using `pkExec` now.